src/parser/ParsedQuery.h

```diff
@@ -327,9 +327,32 @@ class ParsedQuery {
   string _textLimit;
   std::optional<size_t> _offset = std::nullopt;
   string _originalString;
-  // explicit default initialisation because constructor is private
+  // explicit default initialisation because the constructor
+  // of SelectClause is private
   std::variant<SelectClause, ConstructClause> _clause{SelectClause{}};
 
+  [[nodiscard]] bool hasSelectClause() const {
+    return std::holds_alternative<SelectClause>(_clause);
+  }
+
+  [[nodiscard]] bool hasConstructClause() const {
+    return std::holds_alternative<ConstructClause>(_clause);
+  }
+
+  [[nodiscard]] decltype(auto) selectClause() const {
+    return std::get<SelectClause>(_clause);
+  }
+
+  [[nodiscard]] decltype(auto) constructClause() const {
+    return std::get<ConstructClause>(_clause);
+  }
+
+  decltype(auto) selectClause() { return std::get<SelectClause>(_clause); }
```

Comment:
The non-const overloads can and probably should also be nodiscard,
because they also have no side-effects (although they don't automatically turn yellow in CLion)